community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -525,9 +524,8 @@
                   "JVM enough heap to hold all your transaction state and query context, and then leave the rest for " +
                   "the page cache. If no page cache memory is configured, then a heuristic setting is computed based " +
                   "on available system resources." )
-    public static final Setting<Long> pagecache_memory =
-            buildSetting( "dbms.memory.pagecache.size", BYTES, null)
-                    .constraint( min( PageCache.PAGE_SIZE * 30L ) ).build();
+    public static final Setting<String> pagecache_memory =
+            buildSetting( "dbms.memory.pagecache.size", STRING, null ).build();
```

Comment:
Is it ok to change property surface 3.3 > 3.4?

Reply:
I'm not sure. The only public API usage of Setting that I can think of is `GraphDatabaseBuilder.setConfig`, where the field is just passed through. The alternative is to have two settings that does the same thing but differ in type. I don't know if they can both be called `dbms.memory.pagecache.size` in that case. If they can't, then I suspect it would cause more confusion.